lib/helpers/reporter.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+/**
+ * Build a Pa11y reporter.
+ *
+ * Same as 'pa11y/lib/reporter' but reporter methods accept multiple arguments
+ * @private
+ * @param {Object} methods - The reporter methods.
+ * @returns {Object} Returns pa11y reporter.
+ */
+module.exports = function buildReporter(methods) {
+	const reporter = {
+		report: new Map(),
+		supports: methods.supports,
+		beforeAll: buildReporterMethod(methods.beforeAll),
+		afterAll: buildReporterMethod(methods.afterAll),
+		begin: buildReporterMethod(methods.begin),
+		results: buildReporterMethod(methods.results),
```

Comment:
Propose adding an `error` method, which the `pa11y` reporters implement:

```suggestion
		results: buildReporterMethod(methods.results),
		error: buildReporterMethod(methods.error),
```